extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/dart/controller/DartTableInputSpecSlicer.java

```diff
@@ -228,27 +282,28 @@ static Set<DartQueryableSegment> findQueryableDataSegments(
    *
    * @throws IllegalStateException if any provided segments do not match the provided datasource
    */
-  static List<InputSlice> makeSegmentSlices(
+  private List<InputSlice> makeSegmentSlices(
       final String dataSource,
-      final List<List<DartQueryableSegment>> assignments
+      final List<DartServerAssignment> assignments
   )
   {
     final List<InputSlice> retVal = new ArrayList<>(assignments.size());
 
-    for (final List<DartQueryableSegment> assignment : assignments) {
+    for (final DartServerAssignment assignment : assignments) {
       if (assignment == null || assignment.isEmpty()) {
         retVal.add(NilInputSlice.INSTANCE);
       } else {
-        final List<RichSegmentDescriptor> descriptors = new ArrayList<>();
-        for (final DartQueryableSegment segment : assignment) {
-          if (!dataSource.equals(segment.getSegment().getDataSource())) {
-            throw new ISE("Expected dataSource[%s] but got[%s]", dataSource, segment.getSegment().getDataSource());
-          }
-
-          descriptors.add(toRichSegmentDescriptor(segment));
-        }
-
-        retVal.add(new SegmentsInputSlice(dataSource, descriptors, ImmutableList.of()));
+        final List<RichSegmentDescriptor> descriptors = assignment.getDartQueryableSegments()
```

Comment:
style nit: this is actually longer (more lines) than the original imperative code that didn't use streams. It's also intended far to the right, which leads to very long lines. IMO, the original imperative code was better stylistically.

Reply:
Makes sense. I went through a few different iterations of the code here, so I forgot the original code by the end.